### PR TITLE
Adds package parameter to use_template()

### PR DESCRIPTION
**Pull request by ijlyttle**
Addresses #120 (but not fixed just yet)

This initial PR is simply to get things started. Will signal when I think it's ready (should not take too long)

### CODE REVIEW

**Comment by ijlyttle** [comment]
I think this is enough to try this out myself to see how it works. Assuming everything works OK, I think it remains to: 

[ ] remove `@keywords internal`
[ ] Add some text to the roxygen documentation
[ ] Add a bullet item to NEWS.md

I'm sure you will add to this list as needed

**Comment by jennybc** [comment]
@ijlyttle When it's convenient, will you rebase/merge? The most important conflict-causing change is that template text is handed around as a character vector now, as opposed to a collapsed string. Also, I apologize in advance that you might need to rebase/merge again before we're done. We're pushing to release shortly after the CRAN holiday, so lots of little changes are happening.

I have some questions about intended usage that I'll put in #120.

**Comment by jennybc** [comment]
Presumably you've done a little manual testing, which looks like the most practical option here. Could add some code/demo for that to this thread?

**Comment by ijlyttle** [comment]
Hi @jennybc,

I have made the rebase/merge but I have a question for you: is there something I could/should do to clean up the git history?

Assuming all goes well with the automated checks, I will test my application again then write some motivation and demo code in this thread. After I make another cup of coffee.

**Comment by jennybc** [comment]
> is there something I could/should do to clean up the git history?

No, don't worry about that for our sake. It will be a squash merge anyway.

**Comment by jennybc** [comment]
OK I did clean up the history. I can actually push into your PR branch. You can add a commit adding a bullet to NEWS, for example, and then when I squash, it will correctly reflect your authorship.

File: R/github.R [hunk 1]
```diff
@@ -112,6 +115,7 @@ use_github <- function(organisation = NULL,
     https = create$clone_url,
     ssh = create$ssh_url
   )
+  print(origin_url)
```
Comment (jennybc) [inline R/github.R]: Should this be removed?

Comment (ijlyttle) [reply R/github.R]: oh dear - it should - on it!

**Comment by jennybc** [comment]
If you polish off the checklist you made above, I think this is ready to merge. Remember to include the "fixes" language also for #164 and #116 that got swept up here.

**Comment by ijlyttle** [comment]
Hi @jennybc: as you asked above, here is some motivation for the overall pull request.

I am putting together a package called [ghentr](https://github.com/ijlyttle/ghentr) aimed towards people who are building institution-specific packages, where the institution has an instance of Github Enterprise (GHE). I hope anyone else reading will forgive me for tailoring an example to you and UBC.

The goal of ghentr is to make it easier for *you*, the power R-user of UBC's GHE, to create a package to that lets your colleagues access UBC's GHE as easily as they access github.com.

You, as a ghentr user, would be creating an R-package of tools specific to UBC. While working on that package, you would call:

```R 
ghentr::use_github_enterprise(
  host = "github.ubc.ca/api/v3",
  suffix = "ubc",
  name = "UBC" # 
)
```

This would create some files in your `R` directory that would contain the following functions:

```
use_github_ubc()
install_github_ubc()
github_ubc_pat()
```

These functions would wrap their usethis and devtools counterparts, setting the default host and authorization according to the UBC particulars you provided.

To do this, ghentr needs `usethis::use_template()` to be exported, which is the motivation behind this PR. It will also help if `code_block()`, `todo()`, and `project_name()` can be exported - this is the motivation behind #126. (if need be, I could copy the styling code into ghentr and add @hadley as a contributor (with his permission) - both are GPL-3)

Just a "few short words"  to put this issue into context.

I'll assume that you want to keep `use_template()` as an internal function, add an item to NEWS, push, then add a comment here with all the fix numbers.

Thanks!

**Comment by ijlyttle** [comment]
Fixes #120, fixes #116, fixes #164 

Happy to deal with #126 in a different PR, if needed. 

Ready-to-go from my perspective. Thanks for all your help 🎉

**Comment by ijlyttle** [comment]
Hi @jennybc, I know you are aiming to get this to CRAN in short order; sorry to be bugging you. Is there anything I can do to help get this PR in that release? 

I understand one of the timetable motivations is rstudio::conf - I have put myself in a similar boat, with [ghentr]() (which I will talk about there) relying on this PR to use templates from other packages.

Thanks again!

**Comment by jennybc** [comment]
@ijlyttle Oh I just got stuck in on different things. I think this is an easy merge once I sit down to do exactly that. And yes we're in same boat re: rstudio::conf!

**Comment by jennybc** [comment]
I guess the one thing that does make it faster: I was hoping for code snippet to make a package and stick a template in it. Then, from another package, use that template. Basically one little example I could quickly run to see the feature in action. I can write that myself but if it pre-exists, even better!

**Comment by ijlyttle** [comment]
@jennybc I think that most of what you are asking for exists with ghentr (see above)

```R
devtools::install_github("r-lib/usethis", ref = "85bb26a")
devtools::install_github("ijlyttle/ghentr")

usethis::create_package("~/temp/ubctools") # tune to your liking, please don't burn my computer 😀
```

At this point (as you know) new RStudio instance opens for **ubctools**. Resuming there:

```R
ghentr::use_github_enterprise(
  host = "github.ubc.ca/api/v3", # taking a guess here
  suffix = "ubc",
  name = "UBC" # or longer name, if you like
)
```

This will write three files into your new project (as well as write a bunch of stuff to the console); the new file `utils-use_ghe.R` is interpolated from this:

https://github.com/ijlyttle/ghentr/blob/master/inst/templates/use_ghe.R

If this is not what you are looking for, please let me know.

**Comment by jennybc** [comment]
Thanks!

**Comment by ijlyttle** [comment]
Thank you! 🎉

**Pull request merged**